reactor-netty-http/src/main/java/reactor/netty/http/server/AbstractHttpServerMetricsHandler.java

```diff
@@ -93,27 +97,28 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
 			dataSentTime = System.nanoTime();
 		}
 
-		if (msg instanceof ByteBufHolder) {
-			dataSent += ((ByteBufHolder) msg).content().readableBytes();
-		}
-		else if (msg instanceof ByteBuf) {
-			dataSent += ((ByteBuf) msg).readableBytes();
-		}
+        dataSent += extractProcessedDataFromBuffer(msg);
 
 		if (msg instanceof LastHttpContent) {
 			promise.addListener(future -> {
-				ChannelOperations<?, ?> channelOps = ChannelOperations.get(ctx.channel());
-				if (channelOps instanceof HttpServerOperations) {
-					HttpServerOperations ops = (HttpServerOperations) channelOps;
-					recordWrite(ops, uriTagValue == null ? ops.path : uriTagValue.apply(ops.path),
-							ops.method().name(), ops.status().codeAsText().toString());
-					if (!ops.isHttp2() && ops.hostAddress() != null) {
-						// This metric is not applicable for HTTP/2
-						// ops.hostAddress() == null when request decoding failed, in this case
-						// we do not report active connection, so we do not report inactive connection
-						recordInactiveConnection(ops);
+				try {
+					ChannelOperations<?, ?> channelOps = ChannelOperations.get(ctx.channel());
+					if (channelOps instanceof HttpServerOperations) {
+						HttpServerOperations ops = (HttpServerOperations) channelOps;
+						recordWrite(ops, uriTagValue == null ? ops.path : uriTagValue.apply(ops.path),
+								ops.method().name(), ops.status().codeAsText().toString());
+						if (!ops.isHttp2() && ops.hostAddress() != null) {
+							// This metric is not applicable for HTTP/2
+							// ops.hostAddress() == null when request decoding failed, in this case
+							// we do not report active connection, so we do not report inactive connection
+							recordInactiveConnection(ops);
+						}
 					}
 				}
+				catch (RuntimeException e) {
+					log.warn("Exception caught while recording metrics.", e);
+					// Allow request-response exchange to continue, unaffected by metrics problem
+				}
```

Comment:
We have reactor.netty.http.HttpMetricsHandlerTests#testServerConnectionsMicrometer failing on the CI

The problem is that `uriTagValue` throws an exception which fails both recordings `recordWrite` and `recordInactiveConnection`.

Should we leave it like this or have a try/catch per recording, basically `recordInactiveConnection` doesn't depend on `uriTagValue` and can provide correct metric if we introduce separate try/catch, otherwise both metrics will be incorrect.

Reply:
Added this back